### PR TITLE
Add ensure-storybook-title

### PR DESCRIPTION
**Pull request by DamienCassou**
This change is [Reviewable](https://reviewable.io/reviews/foretagsplatsen/ftgp-eslint/40)

### CODE REVIEW

File: lib/rules/ensure-storybook-title.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			ExportDefaultDeclaration: function({ declaration }) {
+				let root = getRootOption(context.options)
+				if (!declaration || !declaration.type === "ObjectExpression") {
+					return;
+				}
+
+				let titleProperty = declaration.properties.find((property) => property.key?.name === "title");
+
+				if (!titleProperty) {
+					return;
```
Comment (DamienCassou) [inline lib/rules/ensure-storybook-title.js:55]: maybe we should raise an error here

File: lib/rules/ensure-storybook-title.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+				}
+
+				let { value } = titleProperty.value;
```
Comment (DamienCassou) [inline lib/rules/ensure-storybook-title.js:58]: I would rename `value` to `actualTitle` or just `title`

File: lib/rules/ensure-storybook-title.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+				let filename = context.getFilename();
+
+				let [, expectedTitle] = filename.split(root);
+				expectedTitle = sanitize(expectedTitle);
```
Comment (DamienCassou) [inline lib/rules/ensure-storybook-title.js:62]: I would extract the 3 lines above into a `getExpectedTitle(context)`

File: tests/lib/rules/ensure-storybook-title.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+ruleTester.run("ensure-storybook-title", rule, {
+
+	valid: [
+		{
+			code: `export default {
+\tcomponent: ForecastCellInfoWidget,
```
Comment (DamienCassou) [inline tests/lib/rules/ensure-storybook-title.js:27]: I would remove the `component` property in all but 1 example as this is unrelated and polluting

File: tests/lib/rules/ensure-storybook-title.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+\ttitle: "accounting/widgets/ForecastCellInfoWidget",
+};`,
+			filename: "fooboo/Client/js/accounting/widgets/ForecastCellInfoWidget/ForecastCellInfoWidget.js",
+			parserOptions: { ecmaVersion: 6, sourceType: "module" },
```
Comment (DamienCassou) [inline tests/lib/rules/ensure-storybook-title.js:31]: I would extract the right part into a `parserOptions` variable

File: tests/lib/rules/ensure-storybook-title.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+		},
+		{
+			code: `export default {
+\tcomponent: ForecastCellInfoWidget,
+};`,
+			filename: "fooboo/Client/js/accounting/widgets/ForecastCellInfoWidget/ForecastCellInfoWidget.js",
+			parserOptions: { ecmaVersion: 6, sourceType: "module" },
+		}
```
Comment (DamienCassou) [inline tests/lib/rules/ensure-storybook-title.js:39]: I would add another example with a non-default value of the `root` option

File: lib/rules/ensure-storybook-title.js (hunk 3 above)
Comment (BenjaminVanRyseghem) [reply lib/rules/ensure-storybook-title.js:62]: done

File: lib/rules/ensure-storybook-title.js (hunk 2 above)
Comment (BenjaminVanRyseghem) [reply lib/rules/ensure-storybook-title.js:58]: done

File: lib/rules/ensure-storybook-title.js (hunk 1 above)
Comment (BenjaminVanRyseghem) [reply lib/rules/ensure-storybook-title.js:73]: done

**Review by BenjaminVanRyseghem — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/foretagsplatsen/ftgp-eslint/40)* status: 0 of 3 files reviewed, 6 unresolved discussions (waiting on @BenjaminVanRyseghem and @DamienCassou)

___
*[tests/lib/rules/ensure-storybook-title.js, line 31 at r1](https://reviewable.io/reviews/foretagsplatsen/ftgp-eslint/40#-MwC-lA6okL6CUVGY_r3-r1-31:-MwCC3X6Ax5HE4Vj95eR:bb7f2mf) ([raw file](https://github.com/foretagsplatsen/ftgp-eslint/blob/dd6c5e7434e7563be11c14e4a03b52ffdc72c67e/tests/lib/rules/ensure-storybook-title.js#L31)):*
<details><summary><i>Previously, DamienCassou (Damien Cassou) wrote…</i></summary><blockquote>

I would extract the right part into a `parserOptions` variable
</blockquote></details>

done
___
*[tests/lib/rules/ensure-storybook-title.js, line 39 at r1](https://reviewable.io/reviews/foretagsplatsen/ftgp-eslint/40#-MwC-lA6okL6CUVGY_r3-r1-39:-MwCC41W61ePIzyRpbsL:bb7f2mf) ([raw file](https://github.com/foretagsplatsen/ftgp-eslint/blob/dd6c5e7434e7563be11c14e4a03b52ffdc72c67e/tests/lib/rules/ensure-storybook-title.js#L39)):*
<details><summary><i>Previously, DamienCassou (Damien Cassou) wrote…</i></summary><blockquote>

I would add another example with a non-default value of the `root` option
</blockquote></details>

done

File: lib/rules/ensure-storybook-title.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+	return string;
+}
+
+function sanitize(string) {
+	let value = trimStart(string);
+	return path.dirname(value);
+}
+
+function getExpectedTitle({ context }) {
```
Comment (DamienCassou) [inline lib/rules/ensure-storybook-title.js:41]: `{context}` => `context`
